src/time_integration/paired_explicit_runge_kutta/methods_PERK3.jl

```diff
@@ -0,0 +1,386 @@
+function c_PERK3_SSP33(num_stages, c_s2)
+    c = zeros(num_stages)
+
+    # Last timesteps as for SSPRK33
+    c[num_stages] = 0.5
+    c[num_stages - 1] = 1
+
+    # Linear increasing timestep for remainder
+    for i in 2:(num_stages - 2)
+        c[i] = c_s2 * (i - 1) / (num_stages - 3)
+    end
+
+    return c
+end
+
+function PERK3_butcher_tableau_objective_function(a_unknown, num_stages,
+                                                  num_stage_evals,
+                                                  mon_coeffs, c_s2)
+    c_ts = c_PERK3_SSP33(num_stages, c_s2) # ts = timestep
+
+    c_eq = zeros(num_stage_evals - 2) # Add equality constraint that c_s2 is equal to 1
+    # Both terms should be present
+    for i in 1:(num_stage_evals - 4)
+        term1 = a_unknown[num_stage_evals - 1]
+        term2 = a_unknown[num_stage_evals]
+        for j in 1:i
+            term1 *= a_unknown[num_stage_evals - 1 - j]
+            term2 *= a_unknown[num_stage_evals - j]
+        end
+        term1 *= c_ts[num_stages - 2 - i] * 1 / 6
+        term2 *= c_ts[num_stages - 1 - i] * 4 / 6
+
+        c_eq[i] = mon_coeffs[i] - (term1 + term2)
+    end
+
+    # Highest coefficient: Only one term present
+    i = num_stage_evals - 3
+    term2 = a_unknown[num_stage_evals]
+    for j in 1:i
+        term2 *= a_unknown[num_stage_evals - j]
+    end
+    term2 *= c_ts[num_stages - 1 - i] * 4 / 6
+
+    c_eq[i] = mon_coeffs[i] - term2
+    c_eq[num_stage_evals - 2] = 1.0 - 4 * a_unknown[num_stage_evals] -
+                                a_unknown[num_stage_evals - 1]
+
+    return c_eq
+end
+
+function compute_PERK3_butcher_tableau(num_stages, tspan, eig_vals::Vector{ComplexF64},
+                                       c_s2)
+    # Initialize array of c
+    c = c_PERK3_SSP33(num_stages, c_s2)
+
+    # Initialize the array of our solution
+    a_unknown = zeros(num_stages)
+
+    # Special case of e = 3
+    if num_stages == 3
+        a_unknown = [0, c[2], 0.25]
+
+    else
+        # Calculate coefficients of the stability polynomial in monomial form
+        cons_order = 3
+        dtmax = tspan[2] - tspan[1]
+        dt_eps = 1e-9
+        filter_threshold = 1e-12
+
+        num_eig_vals, eig_vals = filter_eigvals(eig_vals, filter_threshold)
+
+        mon_coeffs, dt_opt = bisection(cons_order, num_eig_vals, num_stages, dtmax,
+                                       dt_eps, eig_vals)
+        mon_coeffs = undo_normalization!(cons_order, num_stages, mon_coeffs)
+
+        # Define the objective_function
+        function objective_function(x)
+            return PERK3_butcher_tableau_objective_function(x, num_stages, num_stages,
+                                                            mon_coeffs, c_s2)
+        end
+
+        # Call nlsolver to solve repeatedly until the result is not NaN or negative values
+        is_sol_valid = false
+        while !is_sol_valid
+            # Initialize initial guess
+            x0 = 0.1 .* rand(num_stages)
```

Comment:
This might lead to troubles at testing, maybe we need to change this to something fixed. But let's see.